packages/clerk-js/src/core/auth/AuthCookieService.ts

```diff
@@ -0,0 +1,152 @@
+
+import { clerkCoreErrorTokenRefreshFailed, clerkMissingDevBrowserJwt } from '../errors';
+import { eventBus, events } from '../events';
+import type { FapiClient } from '../fapiClient';
+import type { ClientUatCookieHandler } from './cookies/clientUat';
+import { createClientUatCookie } from './cookies/clientUat';
+import type { SessionCookieHandler } from './cookies/session';
+import { createSessionCookie } from './cookies/session';
+import type { DevBrowser } from './devBrowser';
+import { createDevBrowser } from './devBrowser';
+import { SessionCookiePoller } from './SessionCookiePoller';
+
+// TODO(@dimkl): make AuthCookieService singleton since it handles updating cookies using a poller
+// and we need to avoid updating them concurrently.
+/**
+ * The AuthCookieService class is a service responsible to handle
+ * all operations and helpers required in a standard browser context
+ * based on the cookies to remove the dependency between cookies
+ * and auth from the Clerk instance.
+ * This service is responsible to:
+ *   - refresh the session cookie using a poller
+ *   - refresh the session cookie on tab visibility change
+ *   - update the related cookies listening to the `token:update` event
+ *   - initialize auth related cookies for development instances (eg __client_uat, __clerk_db_jwt)
+ *   - cookie setup for production / development instances
+ * It also provides the following helpers:
+ *   - isSignedOut(): check if the current user is signed-out using cookies
+ *   - urlWithAuth(): decorates url with auth related info (eg dev browser jwt)
+ *   - handleUnauthenticatedDevBrowser(): resets dev browser in case of invalid dev browser
+ *   - setEnvironment(): update cookies (eg client_uat) related to environment
+ */
+export class AuthCookieService {
+  private environment: EnvironmentResource | undefined;
+  private poller: SessionCookiePoller | null = null;
+  private clientUat: ClientUatCookieHandler;
+  private sessionCookie: SessionCookieHandler;
+  private devBrowser: DevBrowser;
+
+  constructor(private clerk: Clerk, fapiClient: FapiClient) {
+    // set cookie on token update
+    eventBus.on(events.TokenUpdate, ({ token }) => {
+      this.updateSessionCookie(token && token.getRawString());
+      this.setClientUatCookieForDevelopmentInstances();
+    });
+
+    this.refreshTokenOnVisibilityChange();
+    this.startPollingForToken();
+
+    this.clientUat = createClientUatCookie();
+    this.sessionCookie = createSessionCookie();
+    this.devBrowser = createDevBrowser({
+      frontendApi: clerk.frontendApi,
+      fapiClient,
+    });
+  }
+
+  // TODO(@dimkl): Replace this method call with an event listener to decouple Clerk with setEnvironment
+  public setEnvironment(environment: EnvironmentResource) {
+    this.environment = environment;
+    this.setClientUatCookieForDevelopmentInstances();
+  }
+
+  public isSignedOut() {
+    return this.clientUat.get() <= 0;
+  }
```

Comment:
@nikosdouvlis what's the "normal" way to check for signed out?

Reply:
In clerk-js/clerk-react the usual way to do this would be `!!Clerk.user` I believe. 
clerk-js has a mutable state so we've never had to check a *cookie* for the signed in status. Since we have an in-memory state, I don't think we should rely to the cookie for this.

Could it by that the `isSignedOut` method name is confusing here? 

I see that before this PR, the caller used to do: 
```
  #shouldSyncWithPrimary = (): boolean => {
    if (this.#hasJustSynced()) {
      this.#clearJustSynced();
      return false;
    }

    if (!this.isSatellite) {
      return false;
    }

    return getClientUatCookie() <= 0;
  };
```

If this is the only usage, what if we changed the name to something mre specific?

Reply:
The cookie is being used here because the `#shouldSyncWithPrimary()` is being called before the ClerkJS is loaded so there is no access to the user.
We could enhance the `isSignedOut()` method to cover both cases or we could rename it. @nikosdouvlis What would you prefer? If we decide to rename it, could you suggest an alternative name?
```typescript
  public isSignedOut() {
    if (!this.clerk.loaded) {
      return this.clientUat.get() <= 0;
    }
    return !!this.clerk.user;
  }
  ```

Reply:
Talked with Dimitris offline and it seems that my concerns where not valid, lets go with the initial proposal :)